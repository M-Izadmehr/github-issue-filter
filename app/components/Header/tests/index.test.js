import React from 'react';
import { shallow } from 'enzyme';

import { Header } from '../index';

describe('Issues Page', () => {
  const fakeData = {
    route: '',
  };

  const fakeDataSearchPage = { ...fakeData, route: '/search/react' };
  it('should render the header', () => {
    const renderedComponent = shallow(
      <Header {...fakeData} />,
    );
    expect(renderedComponent).toMatchSnapshot();
  });
  it('should render the when in search apge', () => {
    const renderedComponent = shallow(
      <Header {...fakeDataSearchPage} />,
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
