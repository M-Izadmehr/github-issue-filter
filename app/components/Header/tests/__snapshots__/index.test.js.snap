// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issues Page should render the header 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="comp-header"
>
    <nav
        className="navbar navbar-light bg-light justify-content-between"
    >
        <a
            className="navbar-brand"
            href="/"
        >
            Issue Manager
        </a>
        <form
            className="form-inline"
        >
            <input
                aria-label="Search"
                className="form-control mr-sm-2"
                onChange={[Function]}
                placeholder="Search for a repository"
                type="search"
                value=""
            />
            <Link
                replace={false}
                to="/search/"
            >
                <button
                    className="btn btn-outline-success my-2 my-sm-0"
                    type="submit"
                >
                    Search
                </button>
            </Link>
        </form>
    </nav>
</div>,
  "nodes": Array [
    <div
      className="comp-header"
>
      <nav
            className="navbar navbar-light bg-light justify-content-between"
      >
            <a
                  className="navbar-brand"
                  href="/"
            >
                  Issue Manager
            </a>
            <form
                  className="form-inline"
            >
                  <input
                        aria-label="Search"
                        className="form-control mr-sm-2"
                        onChange={[Function]}
                        placeholder="Search for a repository"
                        type="search"
                        value=""
                  />
                  <Link
                        replace={false}
                        to="/search/"
                  >
                        <button
                              className="btn btn-outline-success my-2 my-sm-0"
                              type="submit"
                        >
                              Search
                        </button>
                  </Link>
            </form>
      </nav>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Header
        route=""
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "route": "",
        },
        "refs": Object {},
        "state": Object {
          "searchInput": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="comp-header"
>
          <nav
                    className="navbar navbar-light bg-light justify-content-between"
          >
                    <a
                              className="navbar-brand"
                              href="/"
                    >
                              Issue Manager
                    </a>
                    <form
                              className="form-inline"
                    >
                              <input
                                        aria-label="Search"
                                        className="form-control mr-sm-2"
                                        onChange={[Function]}
                                        placeholder="Search for a repository"
                                        type="search"
                                        value=""
                              />
                              <Link
                                        replace={false}
                                        to="/search/"
                              >
                                        <button
                                                  className="btn btn-outline-success my-2 my-sm-0"
                                                  type="submit"
                                        >
                                                  Search
                                        </button>
                              </Link>
                    </form>
          </nav>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="comp-header"
>
          <nav
                    className="navbar navbar-light bg-light justify-content-between"
          >
                    <a
                              className="navbar-brand"
                              href="/"
                    >
                              Issue Manager
                    </a>
                    <form
                              className="form-inline"
                    >
                              <input
                                        aria-label="Search"
                                        className="form-control mr-sm-2"
                                        onChange={[Function]}
                                        placeholder="Search for a repository"
                                        type="search"
                                        value=""
                              />
                              <Link
                                        replace={false}
                                        to="/search/"
                              >
                                        <button
                                                  className="btn btn-outline-success my-2 my-sm-0"
                                                  type="submit"
                                        >
                                                  Search
                                        </button>
                              </Link>
                    </form>
          </nav>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header
    route=""
/>,
}
`;

exports[`Issues Page should render the when in search apge 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="comp-header"
>
    <nav
        className="navbar navbar-light bg-light justify-content-between"
    >
        <a
            className="navbar-brand"
            href="/"
        >
            Issue Manager
        </a>
        <form
            className="form-inline"
        >
            <input
                aria-label="Search"
                className="form-control mr-sm-2"
                onChange={[Function]}
                placeholder="Search for a repository"
                type="search"
                value="react"
            />
            <Link
                replace={false}
                to="/search/react"
            >
                <button
                    className="btn btn-outline-success my-2 my-sm-0"
                    type="submit"
                >
                    Search
                </button>
            </Link>
        </form>
    </nav>
</div>,
  "nodes": Array [
    <div
      className="comp-header"
>
      <nav
            className="navbar navbar-light bg-light justify-content-between"
      >
            <a
                  className="navbar-brand"
                  href="/"
            >
                  Issue Manager
            </a>
            <form
                  className="form-inline"
            >
                  <input
                        aria-label="Search"
                        className="form-control mr-sm-2"
                        onChange={[Function]}
                        placeholder="Search for a repository"
                        type="search"
                        value="react"
                  />
                  <Link
                        replace={false}
                        to="/search/react"
                  >
                        <button
                              className="btn btn-outline-success my-2 my-sm-0"
                              type="submit"
                        >
                              Search
                        </button>
                  </Link>
            </form>
      </nav>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Header
        route="/search/react"
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "route": "/search/react",
        },
        "refs": Object {},
        "state": Object {
          "searchInput": "react",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="comp-header"
>
          <nav
                    className="navbar navbar-light bg-light justify-content-between"
          >
                    <a
                              className="navbar-brand"
                              href="/"
                    >
                              Issue Manager
                    </a>
                    <form
                              className="form-inline"
                    >
                              <input
                                        aria-label="Search"
                                        className="form-control mr-sm-2"
                                        onChange={[Function]}
                                        placeholder="Search for a repository"
                                        type="search"
                                        value="react"
                              />
                              <Link
                                        replace={false}
                                        to="/search/react"
                              >
                                        <button
                                                  className="btn btn-outline-success my-2 my-sm-0"
                                                  type="submit"
                                        >
                                                  Search
                                        </button>
                              </Link>
                    </form>
          </nav>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="comp-header"
>
          <nav
                    className="navbar navbar-light bg-light justify-content-between"
          >
                    <a
                              className="navbar-brand"
                              href="/"
                    >
                              Issue Manager
                    </a>
                    <form
                              className="form-inline"
                    >
                              <input
                                        aria-label="Search"
                                        className="form-control mr-sm-2"
                                        onChange={[Function]}
                                        placeholder="Search for a repository"
                                        type="search"
                                        value="react"
                              />
                              <Link
                                        replace={false}
                                        to="/search/react"
                              >
                                        <button
                                                  className="btn btn-outline-success my-2 my-sm-0"
                                                  type="submit"
                                        >
                                                  Search
                                        </button>
                              </Link>
                    </form>
          </nav>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header
    route="/search/react"
/>,
}
`;
